internal/pkg/git/git.go

```diff
@@ -182,7 +185,7 @@ func (r *Repository) runGitCmd(ctx context.Context, args ...string) (cmdResult,
 	retry := newBackoff()
 	for {
 		result, err := r.doRunGitCmd(ctx, args...)
-		if result.exitCode == 0 {
+		if result.exitCode == 0 && err == nil {
```

Comment:
Mozno sa nejedna o problem git prikazu, napr. sa ani nespusti, a je to nejaka OS level chyba?
Preto som sem doplnil podmienku na retry.